### PR TITLE
Merge REVM upgrades

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Resolves all conflicts over a fairly large refactor in upstream REVM for 4844 and RETH's revm state integration.

### CODE REVIEW

File: crates/revm/src/evm_impl.rs [hunk 1]
```diff
@@ -155,29 +157,33 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
...
-        let tx_caller = env.tx.caller;
-        let tx_data = &env.tx.data;
-        let tx_is_create = env.tx.transact_to.is_create();
-
-        let initial_gas_spend = initial_tx_gas::<GSPEC>(tx_data, tx_is_create, &env.tx.access_list);
+        let initial_gas_spend = initial_tx_gas::<GSPEC>(
+            &env.tx.data,
+            env.tx.transact_to.is_create(),
+            &env.tx.access_list,
+        );
```
Comment (clabby) [inline crates/revm/src/evm_impl.rs:168]: This was the only conflict in the `evm_impl`.

File: crates/primitives/src/specification.rs [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #![allow(non_camel_case_types)]
```
Comment (clabby) [inline crates/primitives/src/specification.rs:1]: The way `SpecId`s are implemented changed, there's now a direct `enabled` fn on the `SpecId` enum itself. I've added support for this and two more tests mirroring the other ones on the `Spec` trait impl.

File: crates/primitives/src/result.rs [hunk 3]
```diff
@@ -122,25 +122,26 @@ impl Output {
```
Comment (clabby) [inline crates/primitives/src/result.rs:122]: Some conflicts in this file; Mostly minimal around 4844 errors etc.

File: crates/primitives/src/env.rs [hunk 4]
```diff
@@ -46,37 +70,124 @@ pub struct OptimismFields {
...
+    /// [EIP-155]: https://eips.ethereum.org/EIPS/eip-155
+    pub chain_id: Option<u64>,
+
+    /// A list of addresses and storage keys that the transaction plans to access.
+    ///
+    /// Added in [EIP-2930].
+    ///
+    /// [EIP-2930]: https://eips.ethereum.org/EIPS/eip-2930
     pub access_list: Vec<(B160, Vec<U256>)>,
+
```
Comment (clabby) [inline crates/primitives/src/env.rs:118]: Conflicts in the env for 4844 support; Not too hard to resolve, new fields for blob txs.

**Review by refcell — APPROVED** [review]
LGTM

**Pull request merged**